frontend/src/pages/FunctionalComponent/RecurringRunDetailsV2FC.tsx

```diff
@@ -0,0 +1,207 @@
+ * limitations under the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { useQuery } from 'react-query';
+import Buttons, { ButtonKeys } from 'src/lib/Buttons';
+import DetailsTable from 'src/components/DetailsTable';
+import { V2beta1Experiment } from 'src/apisv2beta1/experiment';
+import { V2beta1RecurringRun, V2beta1RecurringRunStatus } from 'src/apisv2beta1/recurringrun';
+import { Apis } from 'src/lib/Apis';
+import { PageProps } from 'src/pages/Page';
+import { RoutePage, RouteParams } from 'src/components/Router';
+import { Breadcrumb, ToolbarProps } from 'src/components/Toolbar';
+import { classes } from 'typestyle';
+import { commonCss, padding } from 'src/Css';
+import { KeyValue } from 'src/lib/StaticGraphParser';
+import { formatDateString, enabledDisplayStringV2, errorToMessage } from 'src/lib/Utils';
+import { triggerDisplayString } from 'src/lib/TriggerUtils';
+
+export function RecurringRunDetailsV2FC(props: PageProps) {
+  const { updateBanner, updateToolbar } = props;
+  const [refresh, setRefresh] = useState(true);
+  const [toolbarState, setToolbarState] = useState<ToolbarProps>();
+  const recurringRunId = props.match.params[RouteParams.recurringRunId];
+  const Refresh = () => setRefresh(refreshed => !refreshed);
+
+  const { data: recurringRun, refetch: refetchRecurringRun } = useQuery<
+    V2beta1RecurringRun | undefined,
+    Error
+  >(
+    ['recurringRun'],
+    async () => {
+      try {
+        return await Apis.recurringRunServiceApi.getRecurringRun(recurringRunId);
+      } catch (err) {
+        const errorMessage = await errorToMessage(err);
+        updateBanner({
+          additionalInfo: errorMessage ? errorMessage : undefined,
+          message:
+            `Error: failed to retrieve recurring run: ${recurringRunId}.` +
+            (errorMessage ? ' Click Details for more information.' : ''),
+          mode: 'error',
+        });
+        return;
+      }
+    },
+    { enabled: !!recurringRunId, staleTime: 0, cacheTime: 0 },
+  );
+
+  const experimentId = recurringRun?.experiment_id!;
+  const { data: experiment } = useQuery<V2beta1Experiment | undefined, Error>(
+    ['experiment'],
+    async () => {
+      try {
+        return await Apis.experimentServiceApiV2.getExperiment(experimentId);
+      } catch (err) {
+        const errorMessage = await errorToMessage(err);
+        updateBanner({
+          additionalInfo: errorMessage ? errorMessage : undefined,
+          message:
+            `Error: failed to retrieve this recurring run's experiment.` +
+            (errorMessage ? ' Click Details for more information.' : ''),
+          mode: 'warning',
+        });
+        return;
+      }
+    },
+    { enabled: !!experimentId, staleTime: 0 },
+  );
+
+  useEffect(() => {
+    setToolbarState(getInitialToolbarState());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [recurringRun]);
```

Comment:
Suggestion: Instead of observing the `recurringRun` object, you can actually create a `useState` for `recurringRun.recurring_run_id`. Then update the line 87 to observe the `recurringRunId` state, and update the getInitialToolbarState() to replace `recurringRun.recurring_run_id` with `recurringRunId` state. It might be able to reduce the number of re-rendering caused by recurringRun changes.